### PR TITLE
Ensure .gitignore when setup a new boilerplate

### LINKED ISSUE #323: .gitignore is renamed to .npmignore in boilerplate
This issue from npm was there since 2011 (https://github.com/npm/npm/issues/1862)

For those who wanna contribute to statinamic. This is a easy patch. Just rename .npmignore to .gitignore after boilerplate is copied

Ref: https://github.com/MoOx/statinamic/blob/81baba84c5b8a49564ef8b2c82996148bf8f02ed/src/bin/commands/setup.js#L51-L52

### PR DESCRIPTION
**Pull request by thangngoc89**
Close #323

### CODE REVIEW

File: npm/postinstall.js [hunk 1]
```diff
@@ -2,12 +2,30 @@ const stat = require("fs").stat
 const spawn = require("child_process").spawn
 const join = require("path").join
 const pkg = require("../package.json")
-
+const fs = require("fs-extra")
```
Comment (MoOx) [inline npm/postinstall.js]: can't you use fs-promise here too?

File: npm/postinstall.js [hunk 2]
```diff
@@ -2,12 +2,30 @@ const stat = require("fs").stat
...
 // no need for th is step on CI
 if (process.env.CI) {
   process.exit(0)
 }
 
+const boilerplateDir = join(__dirname, "../boilerplate")
+
+stat(join(boilerplateDir, ".npmignore"), function(err, stats) {
+  if (err || stats.isDirectory()) {
+    return true
```
Comment (MoOx) [inline npm/postinstall.js]: why do you return true here?

Comment (MoOx) [inline npm/postinstall.js]: maybe just if (err) console.log("no npmignore in boilerplate to rename)?

Comment (MoOx) [inline npm/postinstall.js]: no need to check directory

Comment (thangngoc89) [inline npm/postinstall.js]: It's just for escape the function. coding style.

File: npm/postinstall.js (hunk 1 above)
Comment (thangngoc89) [inline npm/postinstall.js]: Why ? fs-promise is just a wrapper of fs-extra. I'm using async operation here :). And I don't like a mix of callback and promise.

**Comment by thangngoc89** [comment]
PR updated

**Comment by MoOx** [comment]
Going to add commit + release as 0.9.1

**Pull request merged**